### PR TITLE
Tac light accessory

### PR DESCRIPTION
**Pull request by SonixApache**
this turns tactical lights into accessories that you can clip onto armored vests and any headgear.

TODO:
 - [x] Add "On" variants for all the sprites
 - [x] Add "On" sprite code
 - [x] Check if the pre-attached sec helmets still work
 - [x] ~~Remove old helmet-with-light sprites~~
 - [x] Override attack so you can only detach it with a screwdriver

🆑 
 - rscadd: Tactical lights are now accessories. You can clip them onto armor and any type of headgear. You'll still need a screwdriver to remove them.

### CODE REVIEW

**Comment by aceedex** [comment]
Why screw driver to remove and not a v*rb?

**Comment by SonixApache** [comment]
Because the old taclights were also only removable by screwdriver.

**Comment by SonixApache** [comment]
Ready for review.

**Comment by ShiftyRail** [comment]
Femsec buff.

**Comment by oversane** [comment]
arie doesn't want to wear the autistic looking helmet does he

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -107,9 +107,30 @@
...
+		var/obj/item/clothing/accessory/taclight/TL = new()
+		if(C.check_accessory_overlap(TL))
+			to_chat(user, "<span class='notice'>You cannot attach more accessories of this type to \the [C].</span>")
+			return
+		if(user.drop_item(src))
+			to_chat(user, "<span class='notice'>You attach \the [src] to \the [C].</span>")
+			TL.source_light = src
+			C.attach_accessory(TL)
+			transfer_fingerprints(src,TL)
+			forceMove(TL)
```
Comment (PaulRitter) [inline code/game/objects/items/devices/flashlight.dm:128]: does just referencing src in TL and then forceMoving TL remove src?

File: code/modules/clothing/accessories/tactical_light.dm [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	icon_state = "taclight"
+	accessory_exclusion = LIGHT
+	var/obj/item/device/flashlight/tactical/source_light
+	ignoreinteract = TRUE
+	
+/obj/item/clothing/accessory/taclight/New()
+	..()
+	if (!source_light)
+		source_light = new /obj/item/device/flashlight/tactical/
+		source_light.forceMove(src)
```
Comment (PaulRitter) [inline code/modules/clothing/accessories/tactical_light.dm:13]: if comment #1 is true, i'd suggest just removing one tab here

**Review by PaulRitter — CHANGES_REQUESTED** [review]

File: code/modules/clothing/accessories/tactical_light.dm [hunk 3]
```diff
@@ -0,0 +1,117 @@
+/obj/item/clothing/accessory/taclight
+	name = "tactical light"
+	desc = "This is attached to something."
+	icon_state = "taclight"
+	accessory_exclusion = LIGHT
+	var/obj/item/device/flashlight/tactical/source_light
```
Comment (ShiftyRail) [inline code/modules/clothing/accessories/tactical_light.dm:6]: Make sure to set this to `null` in `Destroy()`.

File: code/modules/clothing/accessories/tactical_light.dm [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		update_icon()
+		attached_to.update_icon()
+
+/obj/item/clothing/accessory/taclight/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/screwdriver) && attached_to)
+		to_chat(user, "<span class='notice'>You remove [src] from [attached_to].</span>")
+		attached_to.remove_accessory(user, src)	
+		
+/obj/item/clothing/accessory/taclight/on_accessory_interact()
+	return -1
```
Comment (ShiftyRail) [inline code/modules/clothing/accessories/tactical_light.dm]: Magic numbers, add a comment explaining what `-1` means.

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -30,7 +30,7 @@
 		accessory.emp_act(severity)
 	..()
 
-/obj/item/clothing/attackby(obj/item/I, mob/user)
+/obj/item/clothing/attackby(obj/item/I as obj, mob/user as mob)
```
Comment (ShiftyRail) [inline code/modules/clothing/clothing.dm]: `as x` are completely unneeded outside of verbs. Remove it.

File: code/modules/clothing/clothing.dm [hunk 6]
```diff
@@ -47,6 +47,10 @@
 			var/mob/living/carbon/human/H = loc
 			H.update_inv_by_slot(slot_flags)
 		return 1
+	if(istype(I, /obj/item/weapon/screwdriver))
```
Comment (ShiftyRail) [inline code/modules/clothing/clothing.dm]: Prefer `I.is_screwdriver(user)` for the new proc that makes coins act as screwdrivers.

File: code/modules/clothing/clothing.dm [hunk 7]
```diff
@@ -64,9 +68,14 @@
 					delayed.Add(A)
 				else
 					continue
+		var/ignorecounter = 0
 		for(var/obj/item/clothing/accessory/A in delayed)
-			if(A.on_accessory_interact(user, 1))
+			if(A.ignoreinteract)
+				ignorecounter += 1
```
Comment (ShiftyRail) [inline code/modules/clothing/clothing.dm]: `ignorecounter += A.ignoreinteract` ? It condensates the two ones lines one.

**Review by ShiftyRail — CHANGES_REQUESTED** [review]
Just a quick review

File: code/modules/clothing/clothing.dm (hunk 6 above)
Comment (SonixApache) [reply code/modules/clothing/clothing.dm]: Can't figure out how to get that to work with this.

Comment (PaulRitter) [reply code/modules/clothing/clothing.dm]: I.is_screwdriver(user) will return 1 if its a screwdriver, so replace `if(istype(I, /obj/item/weapon/screwdriver))` with `if(I.is_screwdriver(user))`

the user param is relevant for cases where the obj would only be considered a screwdriver when on harm intent(or other user related things) eg. coins

Comment (SonixApache) [reply code/modules/clothing/clothing.dm]: Yeah, sadly, I doesn't support `is_screwdriver(user)` in this case. It is a proc that does not exist.

```
/obj/item/clothing/accessory/taclight/attackby(var/obj/item/I, var/mob/user)
    if(I.is_screwdriver(user) && attached_to)
        attached_to.remove_accessory(user, src)    

code\modules\clothing\accessories\tactical_light.dm:97:error: I.is_screwdriver: undefined proc
```

I suppose I could go around it by turning the typecheck into an `obj/item/tool` first, but it all seems kind of convoluted.

Comment (SonixApache) [reply code/modules/clothing/clothing.dm]: i forgot to update my code and exxion told me to update my code and now the proc works thank you and goodbye

Comment (PaulRitter) [reply code/modules/clothing/clothing.dm]: u a silli boi

**Review by ShiftyRail — APPROVED** [review]

**Comment by Blithering** [comment]
>Removed the glowing eyes for the HoS dermal implant

REEEEEEEEEEEEEEEEEEE
reee

**Comment by SonixApache** [comment]
just code glowing eye implants dude

**Pull request merged**